capa/features/extractors/ghidra/file.py

```diff
@@ -0,0 +1,198 @@
+        off, mzx, pex, i = todo.pop()
+
+        # MZ header has one field we will check e_lfanew is at 0x3c
+        e_lfanew = off.add(0x3C)
+
+        if seg_max.getOffset() < (e_lfanew.getOffset() + 4):
+            continue
+
+        e_lfanew_bytes = b''
+        e_lfanew_sbytes = getBytes(e_lfanew, 4) 
+        for b in e_lfanew_sbytes:
+            b = (b & 0xFF).to_bytes(1, 'little')
+            e_lfanew_bytes = e_lfanew_bytes + b
+        newoff = struct.unpack("<I", capa.features.extractors.helpers.xor_static(e_lfanew_bytes, i))[0]
+
+        # assume XOR'd "PE" bytes exist within threshold
+        if newoff > MAX_OFFSET_PE_AFTER_MZ:
+            continue
+
+        peoff = off.add(newoff)
+        if seg_max.getOffset() < (peoff.getOffset() + 2):
+            continue
+
+        if getBytes(peoff, 2) == pex:
+            yield off, i
+
+
+def extract_file_embedded_pe() -> Iterator[Tuple[Feature, Address]]:
+    """extract embedded PE features
+    """
+    for seg in currentProgram.getMemory().getBlocks():
+        for ea, _ in check_segment_for_pe(seg):
+            yield Characteristic("embedded pe"), FileOffsetAddress(ea)
+
+
+def extract_file_export_names() -> Iterator[Tuple[Feature, Address]]:
+    """extract function exports
+
+    Exports in Ghidra:
+        - namespace = Global
+        - SourceType = IMPORTED
+    """
+
+    fm = currentProgram.getFunctionManager().getFunctions(True) # Only Global functions
+    while fm.hasNext():
+        f = fm.next().getSymbol()
+        if (f.getSource() == SourceType.IMPORTED):
+            yield Export(f.toString()), AbsoluteVirtualAddress(f.getAddress().getOffset())
+
+
+def extract_file_import_names() -> Iterator[Tuple[Feature, Address]]:
+    """extract function imports
+
+    1. imports by ordinal:
+     - modulename.#ordinal
+
+    2. imports by name, results in two features to support importname-only
+       matching:
+     - modulename.importname
+     - importname
+    """
+
+    fm = currentProgram.getFunctionManager().getExternalFunctions()
+    while fm.hasNext():
+        f = fm.next()
+        addr = int(f.getEntryPoint().toString().split(':')[1], 16)  # format: EXTERNAL:<hex_addr>
+        fstr = f.getName()  # format: 'importname' / 'Ordinal_*'
+        if 'Ordinal_' in fstr:
+            fstr = f"#{fstr.split('_')[1]}"
+        yield Import(fstr), AbsoluteVirtualAddress(addr)
+
+
+def extract_file_section_names() -> Iterator[Tuple[Feature, Address]]:
+    """extract section names"""
+
+    for block in currentProgram.getMemory().getBlocks():
+        yield Section(block.getName()), AbsoluteVirtualAddress(block.getStart().getOffset())
+
+
+def extract_file_strings() -> Iterator[Tuple[Feature, Address]]:
+    """extract ASCII and UTF-16 LE strings"""
+
+    dat = getFirstData()
+    while (dat != None):
+        if (dat.hasStringValue()):
+            yield String(dat.getValue()), FileOffsetAddress(dat.getAddress().getOffset()) 
+        dat = getDataAfter(dat)
```

Comment:
We have had issues in the past with relying on a tool's built-in string identification. e.g. for IDA we opted to manually identify strings for each loaded section: https://github.com/mandiant/capa/blob/2ce4f8769d1f333141be3689a56bd75fa1a66948/capa/features/extractors/ida/file.py#L144-L152

I suggest we do the same here.